### ISSUE TITLE
How to use and set our own credentials with AWSAmplify@3.x.x?

### ISSUE THREAD

**Issue by Jun711** [aws-amplify/amplify-js#8184]
**Is your feature request related to a problem? Please describe.**

I am aware of [federated sign in](https://aws-amplify.github.io/amplify-js/api/classes/authclass.html#federatedsignin). 

However, I would to enable Google sign-in new feature, this [document](https://docs.aws.amazon.com/cognito/latest/developerguide/google.html#set-up-google-1.javascript) suggests to use the following code. 
Since I am already using Amplify for authentication, I would like to continue using it.

However the following code would not work with Amplify@3.x.x. How should I proceed?
```
AWS.credentials and AWS.config don’t exist anymore anywhere in Amplify JavaScript. 
Both options will not be available to use in version 3. You will not be able to use and set your own credentials.
```

```
function signinCallback(authResult) {
  if (authResult['status']['signed_in']) {

     // Add the Google access token to the Amazon Cognito credentials login map.
     AWS.config.credentials = new AWS.CognitoIdentityCredentials({
        IdentityPoolId: 'IDENTITY_POOL_ID',
        Logins: {
           'accounts.google.com': authResult['id_token']
        }
     });

     // Obtain AWS credentials
     AWS.config.credentials.get(function(){
        // Access AWS resources here.
     });
  }
}
```

**Describe the solution you'd like**
Enable using and setting our own credentials.

**Comment by manueliglesias** [comment]
Hi @Jun711 

> I would like to enable Google sign-in new feature that is not supported by federatedSignIn call

Can you elaborate a little on this? what feature that you are looking for is not currently supported?

Also, take a look at the example showing [Identity Pool Federation with google](https://docs.amplify.aws/lib/auth/advanced/q/platform/js#google-sign-in-react), it might help

**Comment by Jun711** [comment]
@manueliglesias 
I am trying to use `Google One Tap Sign-in`. I just found a similar [thread](https://github.com/aws-amplify/aws-sdk-android/issues/1976) opened by another user.

maybe the document you shared may work. I will give it a try.
```
const getAWSCredentials = async (googleUser) => {
        const { id_token, expires_at } = googleUser.getAuthResponse();
        const profile = googleUser.getBasicProfile();
        let user = {
            email: profile.getEmail(),
            name: profile.getName()
        };

        const credentials = await Auth.federatedSignIn(
            'google',
            { token: id_token, expires_at },
            user
        );
        console.log('credentials', credentials);
    }
```

**Comment by Jun711** [comment]
@manueliglesias 

I tried the above code from the document you shared. 

Right after calling federatedSignIn with google id token, I called `await Auth.currentSession` but Amplify log says there is no current user. Amplify debugger log is attached. 

I would like to use the JwtToken to call an API authorized using Cognito. How do I achieve this? 
```
'Authorization': session.getIdToken().getJwtToken(),
```

FederatedSignIn + UserSession code (Amplify part is similar but Google API part is different
```
const getAWSCredentials = async (googleUser) => {
        const { id_token, expires_at } = googleUser.getAuthResponse();
        const profile = googleUser.getBasicProfile();
        let user = {
            email: profile.getEmail(),
            name: profile.getName()
        };

        const credentials = await Auth.federatedSignIn(
            'google',
            { token: id_token, expires_at },
            user
        );
        console.log('credentials', credentials);

        const userSession = await Auth.currentSession();
        console.log('userSession ', userSession)
        const amplifyIdToken = userSession.getIdToken().getJwtToken();
        console.log('amplifyIdToken ', amplifyIdToken)
  }
 ```
 
 
<img width="562" alt="Screen Shot 2021-04-27 at 1 36 03 PM" src="https://user-images.githubusercontent.com/6880939/116309570-c2fba180-a75d-11eb-889f-4d2619ac0b4e.png">

**Comment by manueliglesias** [comment]
Hi @Jun711 

> No Cognito user is created for this user when federatedSignIn this way. Does Amplify provide a way to create an account for this user at my user pool?

Yes, you can use federation as shown in https://docs.amplify.aws/lib/auth/social/q/platform/js

That will use a cognito user pool instead of just identity pools

**Comment by Jun711** [comment]
@manueliglesias 
Yes, I had that setup for traditional Google sign in already and it would create a cognito user when I used Auth.federatedSignIn({provider: 'Google'})}`.

However, the newer feature: Google one tap sign in doesn't have the same configuration for traditional Google sign in as there is no redirection. In other words, it doesn't redirect to Google Sign In page and redirects to my App UI. Could you please look into it further?

And
I am not using `Auth.federatedSignIn({provider: 'Google'})}` 
Instead, I have to use like you suggested last week. 
```
        const credentials = await Auth.federatedSignIn(
            'google',
            { token: id_token, expires_at },
            user
        );
```

**Comment by jysung** [comment]
@Jun711 did you manage to figure out one tap sign in?

**Comment by crowers** [comment]
We also need to support Google One Tap with Cognito. Any solutions anyone? Thanks

**Comment by crowers** [comment]
No news that I'm aware of. Planning to move away from Cognito as the support is so poor

**Comment by tannerabread** [comment]
Hi @m98, @crowers 
We recently discussed this feature request internally and are not sure it is currently possible with how the library is set up. We are still investigating the issue.

**Comment by m98** [comment]
@tannerabread Any chance you can prioritize this?
This is a feature that sooner or later will be requested from you again, and can be seen as a potential limitation of Cognito.

**Comment by tannerabread** [comment]
@m98 You are correct and this is a limitation with Cognito itself. We have communicated the requirement for our service to their team, once it is available from Cognito we can implement in Amplify. I do not currently have a timeline for when the fix will come from their end.

**Comment by crowers** [comment]
> @m98 You are correct and this is a limitation with Cognito itself. We have communicated the requirement for our service to their team, once it is available from Cognito we can implement in Amplify. I do not currently have a timeline for when the fix will come from their end.

Will there be a way to use Google One Tap and Cognito without adding a large library like Amplify to my project?

**Comment by crowers** [comment]
For those planning to migrate away from Cognito we found this blog post outlining the steps to be very helpful: https://fusionauth.io/blog/2022/02/07/how-to-migrate-from-cognito  We hope it will be useful to anyone else similarly struggling with Cognito's limitations.

**Comment by vicodinvic1** [comment]
same problem, our team need this feature as well

**Comment by gabrielmanara** [comment]
Hi @tannerabread, do you have any updates from Cognito team?

**Comment by sjdeak** [comment]
Hi @tannerabread, do you have any updates from Cognito team?

**Comment by pkhadson** [comment]
Hi @tannerabread, do you have any updates from Cognito team?

**Comment by lucasyarid** [comment]
It would be great to have an update on this, even if it is a "won't fix".
It is hard to justify the usage of Cognito when such integrations, which are standard for modern authentication, cannot be implemented.

**Comment by knotri** [comment]
Hi, any update?

**Comment by yousufdev** [comment]
This is crucial, given that almost all authentication providers have it. Any update would be appreciated.

**Comment by devyansh-garg** [comment]
> Hey guys, if you already have google single-sign-in enabled in your client, we found a pretty straightforward way to make One-Tap work quite nicely. You basically just use the One-tap as a "proxy" to make the user select the e-mail address they want to use, and then you trigger the normal google sso flow with that e-mail as the `loginHint`:
> 
> ```
> const oneTapCallback = (oneTapResponse) => {
>       const decodedJWT = decodeJWT(oneTapResponse.credential);
>       const email = decodedJWT.payload.email;
> 
>       if (typeof email !== "string") {
>         throw new Error("Google ID token does not contain an email.");
>       }
> 
>       // once the user has succesfully selected and verified their identity through onetap
>       // take them to the sso flow with the selected email address as the loginHint
>       signInWithRedirect({
>         provider: "Google",
>         options: {
>           loginHint: email,
>         }
>       });
> }
> ```

Thanks you for this @gbyesiltas , helped me a lot.
